sdk/devcenter/Azure.Developer.DevCenter/tests/DevBoxClientTests.cs

```diff
@@ -0,0 +1,415 @@
+
+            string scheduleName = scheduleNameJson.ToString();
+            Assert.AreEqual("default", scheduleName);
+        }
+
+        [RecordedTest]
+        public async Task GetActionsSucceeds()
+        {
+            var numberOfReturnedActions = 0;
+            await foreach (BinaryData actionsData in _devBoxesClient.GetActionsAsync(TestEnvironment.ProjectName, DevBoxName))
+            {
+                numberOfReturnedActions++;
+                JsonElement getActionsResponseData = JsonDocument.Parse(actionsData.ToStream()).RootElement;
+
+                if (!getActionsResponseData.TryGetProperty("name", out var actionNameJson))
+                {
+                    Assert.Fail("The JSON response received from the service does not include the necessary property.");
+                }
+
+                string actionName = actionNameJson.ToString();
+                Assert.AreEqual("schedule-default", actionName);
+            }
+
+            Assert.AreEqual(1, numberOfReturnedActions);
+        }
+
+        [RecordedTest]
+        public async Task GetActionSucceeds()
+        {
+            Response getActionResponse = await _devBoxesClient.GetActionAsync(
+                TestEnvironment.ProjectName,
+                DevBoxName,
+                "schedule-default");
+
+            JsonElement getActionData = JsonDocument.Parse(getActionResponse.ContentStream).RootElement;
+
+            if (!getActionData.TryGetProperty("name", out var actionNameJson))
+            {
+                Assert.Fail("The JSON response received from the service does not include the necessary property.");
+            }
+
+            string actionName = actionNameJson.ToString();
+            Assert.AreEqual("schedule-default", actionName);
+        }
+
+        [RecordedTest]
+        public async Task SkipActionSucceeds()
+        {
+            Response skipActionResponse = await _devBoxesClient.SkipActionAsync(
+                TestEnvironment.ProjectName,
+                DevBoxName,
+                "schedule-default");
+
+            Assert.AreEqual((int)HttpStatusCode.NoContent, skipActionResponse.Status);
+        }
+
+        [RecordedTest]
+        public async Task DelayActionSucceeds()
+        {
+            // Using fixed time to match sessions records
+            string time = "2023-05-02T16:01:53.3821556Z";
+            DateTimeOffset delayUntil = DateTimeOffset.Parse(time);
+
+            Response delayActionResponse = await _devBoxesClient.DelayActionAsync(
+                TestEnvironment.ProjectName,
+                DevBoxName,
+                "schedule-default",
+                delayUntil);
+
+            JsonElement delayActionData = JsonDocument.Parse(delayActionResponse.ContentStream).RootElement;
+            if (!delayActionData.TryGetProperty("next", out var nextActionTimeJson))
+            {
+                Assert.Fail("The JSON response received from the service does not include the necessary property.");
```

Comment:
Do we get the name of the missing property for free via this assertion? Doesn't look like it - might be useful to put the missing property's name in the message to improve diagnosability

Reply:
Fixed: https://github.com/Azure/azure-sdk-for-net/pull/36245/commits/cdee943ada2c3509b2fbe1e097d4d81d86d404c7